### ISSUE TITLE
Screnshot while hyperion active

### ISSUE THREAD

**Issue by joachimpr** [hyperion-project/hyperion#761]
H there,

This is not really an issue buy rather a feature request/reqiest for help. I could not find anywhere else to post. Sorry if it is in the wrong place.

What i would like to do is take a screenshot while hyperion is running or alternatively somehow access the screenshot that hyperion uses to alter the leds while it is actively running. The problem is I cannot take a screenshot with the command line tool while hyperion service is running due to device being busy. What i would like to do is create a seperate python script which analyzes the screenshot an performs an action if certain elements in said screenshot is true. Is this possible or is there somewher specific where the hyperion service saves its screenshot which I can possibly copy to work from? Thank you very much

**Comment by penfold42** [comment]
Hyperion.ng can show the live video with leds overlayed in its web UI.
Maybe you can use its websocket interface from python?

We won't be adding this feature to the old Hyperion code as it's effectively in maintenance mode.

**Comment by penfold42** [comment]
Btw, https://hyperion-project.org/forums/feature-requests.7/ is a good place to discuss further

**Comment by joachimpr** [comment]
That might work thanks. I will just have to figure out how. Is hyperion.ng
a different installation to the current stable version? I will post future
questions on the link provided.

**Comment by penfold42** [comment]
Yes - it's a fresh install and there are no official builds yet so you'll probably have to build from source

**Comment by joachimpr** [comment]
Hi There. I'm back on this project again. I have looked at the hyperion.ng project and I would prefer to go with the current stable release and just add the feature to the source myself. Its rather simple i think. I have looked at the sourcecode but I am a bit confused as to where the grabber is actually grabbing the frame into memory for processing the led lights vs grabbing the screenshot from the remote. All I would like to do is to pick up that frame which is in memory (by dumping it to an image on disk) so that I can process it with a different script. I probably only need 1 every second. So I have 2 questions:

1. Does hyperion save its captures to disk somewhere and if so, where? Or is it all done in memory?

2. If it is all in memory, which part of the source code is responsible for grabbing the frame to memory and processing it for the led lights? Perhaps I can add a step there to just save the frame to disk.

Thanks

**Comment by Paulchen-Panther** [comment]
hi @joachimpr 

1. it is all in memory
2. https://github.com/hyperion-project/hyperion.ng/blob/master/libsrc/hyperion/Hyperion.cpp#L696

**Comment by joachimpr** [comment]
Hi @Paulchen-Panther ,

Sorry to be a bug. Could you direct me to the same line of code but in the current release version of Hyperion at https://github.com/hyperion-project/hyperion?

I have searched through the source code there for SetImage but I find it multiple places. I would just like to add the additional code to the correct place to dump the frame to disk in the one correct place.

Thanks for your help!
